src/snowflake/snowpark/udf.py

```diff
@@ -897,3 +915,115 @@ def _do_register_udf(
                 )
 
         return UserDefinedFunction(func, return_type, input_types, udf_name)
+
+    def __do_register_udf(
```

Comment:
Temporary naming since python does not have overloading feature. Once the new implementation has been reviewed, the older `_do_register_udf` will be removed.